### ISSUE TITLE
about batchnormalization

### ISSUE THREAD

**Issue by wohkqu2008** [phecda-xu/FullyCNNSpeechEnhancement#1]
hi phecda
           Thanks for your sharing of FullyCNN speech enhancement project, I have some questions about structure configuration in file "train.py :" and I post it here as belows:

          **T = 1
        data = Input(shape=[n_freq])
        x = Reshape([1, T, n_freq])(data)
        x1 = Conv2D(10, (T, 11), strides=(10, 1), data_format='channels_first', padding='same')(x)
        x1 = BatchNormalization(axis=-1)(x1)
        x1 = Activation('relu')(x1)**

       my questions are:
1.  since input shape is (1 ,1, n_freq) with data_format='channels_first',, why do we need strides of (10, 1),
       I think it equals setting strides to (1, 1)
2.   have you tried to utilize more temporal information of input features, e.g. input more frames of features rather than one each time
3.   in BatchNormalization, you set axis to -1, that is to say BN will calc mean and variance over elements of each sample excluding the last axis, but  data_format='channels_first', I feel confused.
         Hope get your reply, thanks very much!

**Comment by phecda-xu** [comment]
> hi phecda
> Thanks for your sharing of FullyCNN speech enhancement project, I have some questions about structure configuration in file "train.py :" and I post it here as belows:
> 
> ```
>       **T = 1
>     data = Input(shape=[n_freq])
>     x = Reshape([1, T, n_freq])(data)
>     x1 = Conv2D(10, (T, 11), strides=(10, 1), data_format='channels_first', padding='same')(x)
>     x1 = BatchNormalization(axis=-1)(x1)
>     x1 = Activation('relu')(x1)**
> 
>    my questions are:
> ```
> 
> 1. since input shape is (1 ,1, n_freq) with data_format='channels_first',, why do we need strides of (10, 1),
>    I think it equals setting strides to (1, 1)
> 2. have you tried to utilize more temporal information of input features, e.g. input more frames of features rather than one each time
> 3. in BatchNormalization, you set axis to -1, that is to say BN will calc mean and variance over elements of each sample excluding the last axis, but  data_format='channels_first', I feel confused.
>    Hope get your reply, thanks very much!

Well, just as you can see, i was managed to reproduce the structure, but i don't have enough time to finish it. It's an Incomplete project and have many mistakes. Sorry for that.

Your question are make sense, the paper propose a  1-d Convolution, I made mistakes in there. I think the  Input size of Conv2D layer sould be (batch_size, channels, time_step, freq) , kernel_size should be (T,1), and T>=1, T is 8 in paper. stride can set as (1,1); 

Or maybe use Conv1D layer:

should reshape feature data shape, such as 

![image](https://user-images.githubusercontent.com/46859427/104889146-bc180280-59a8-11eb-8f0f-41475b044867.png)

In paper, set T=8 were considered with speed and effect(T=1 is fastest with bad effect), large T may get better effect.

You can try it by yourself.

**Comment by wohkqu2008** [comment]
hi phecda,
                Thanks very much for your response!
